mempool/src/executor.rs

```diff
@@ -34,6 +35,9 @@ pub(crate) struct MempoolExecutor<N: Network> {
     // Reference to the network, to alow for message validation
     network: Arc<N>,
 
+    // Network ID, used for tx verification
+    network_id: Arc<NetworkId>,
```

Comment:
Mmm... `NetworkId` is an `enum`. I don't think we need to arc it. A copy should work instead

Reply:
That is  because we need to send it to multiple tx verify tasks that are running in parallel...